cdap-ui/app/cdap/main.js

```diff
@@ -208,4 +209,4 @@ CDAP.propTypes = {
   children: PropTypes.node,
 };
 
-ReactDOM.render(<CDAP />, document.getElementById('app-container'));
+ReactDOM.render(<ThemeWrapper render={() => <CDAP />} />, document.getElementById('app-container'));
```

Comment:
the render props have to a function? Can't it simply be `<CDAP />`?

Reply:
Render props are usually function callbacks. I have declared the render prop as a function with ReactNode as return type https://github.com/cdapio/cdap/pull/10924/files#diff-6c54fe7f072e568c645ba781fb1d412eR22.

Do you think we should just have children? Right now have the flexibility of using render props or children. I can remove the render props if you feel it is unnecessary. The idea is for stateless function component we could ideally call the render props callback with the theme object to be directly used.

Reply:
ahh i got things mixed up... I was thinking of modeling React Router, but react router have both render and component. I think this is fine.